sklearn/utils/stats.py

```diff
@@ -15,47 +18,143 @@ def _weighted_percentile(array, sample_weight, percentile=50):
 
     Parameters
     ----------
-    array : 1D or 2D array
+    array : ndarray of shape (n,) or (n, m)
         Values to take the weighted percentile of.
 
-    sample_weight: 1D or 2D array
+    sample_weight: ndarray of (n,) or (n, m)
         Weights for each value in `array`. Must be same shape as `array` or
         of shape `(array.shape[0],)`.
 
-    percentile: int, default=50
+    percentile: inr or float, default=50
         Percentile to compute. Must be value between 0 and 100.
 
+    interpolation : {"linear", "lower", "higher", "nearest"}, default="lower"
+        The interpolation method to use when the percentile lies between
+        data points `i` and `j`:
+
+        * `"linear"`: `i + (j - i) * fraction`, where `fraction` is the
+          fractional part of the index surrounded by `i` and `j`;
```

Comment:
I'm not sure I understand what is the fractional part of an index?

Reply:
Yep it is coming directly from `np.percentile` there they compute the index as a float and check what the fraction to get to `i` and the fraction to get to `j`.

I think that we could describe it on a higher level mentioning that we will linearly interpolate between `i` and `j` using `np.interp`.